cpp/src/column/column.cu

```diff
@@ -176,9 +177,77 @@ struct CreateColumnFromView {
 
 };
 
+struct CreateColumnFromViewVector {
+  std::vector<cudf::column_view> views;
+  cudaStream_t stream;
+  rmm::mr::device_memory_resource *mr;
+
+ template <typename ColumnType,
+           std::enable_if_t<std::is_same<ColumnType, cudf::string_view>::value>* = nullptr>
+ std::unique_ptr<column> operator()() {
+   std::vector<cudf::strings_column_view> sviews;
+   for (auto &v : views) { sviews.emplace_back(v); }
+
+   auto col = cudf::strings::detail::concatenate(sviews, mr, stream);
+
+   cudf::detail::concatenate_masks(views,
+       (col->mutable_view()).null_mask(), stream);
+
+   return std::move(col);
+ }
+
+ template <typename ColumnType,
+           std::enable_if_t<cudf::is_fixed_width<ColumnType>()>* = nullptr>
+ std::unique_ptr<column> operator()() {
+
+   auto type = views.front().type();
+   size_type total_element_count = 0;
+   for (auto &v : views) { total_element_count += v.size(); }
+
+   rmm::device_buffer col_data{
```

Comment:
Why not `rmm::device_vector<ColumnType> col_data{total_element_count}`?

Reply:
I need to move col_data inside a column after I am done copying to it. AFAIK rmm::device_vector like thrust::device_vector does not release ownership of its underlying data to the user. Am I mistaken?

Reply:
That's a good point. `rmm::device_vector` *is* `thrust::device_vector`.